lib/metadata/FrequencyAssembledMetadata.hpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2022 Kotekan Developers
+
+/****************************************************
+ * @file FrequencyAssembledMetadata.hpp
+ * @brief This file declares useful structures and
+ *        the FrequencyAssembledMetadata structure
+ *
+ * @author Mehdi Najafi
+ * @date   28 AUG 2022
+ *****************************************************/
+
+#ifndef FREQ_ASSEMBLED_METADATA
+#define FREQ_ASSEMBLED_METADATA
+
+#include <cstdint> // integer types
+#include <limits>  // std::numeric_limits
+
+/**
+ * @struct FrequencyBin
+ * @brief Frequency bin for a range of frequency ids.
+ * *
+ * @par num_received_frequencies
+ *      The number of frequency ids assigned to the bin
+ * @par lower_band_received_frequency, upper_band_received_frequency
+ *      The frequency id range: [low to high]
+ *
+ * @author Mehdi Najafi
+ */
+struct FrequencyBin {
+    /// The lower frequency range
+    uint32_t lower_band_received_frequency = std::numeric_limits<uint32_t>::max();
+    /// The upper frequency range
+    uint32_t upper_band_received_frequency = std::numeric_limits<uint32_t>::min();
+    /// The number of frequency ids in the frame attached to this metadata
```

Comment:
```suggestion
    /// The lower frequency range
    uint32_t lower_band_received_frequency;
    /// The upper frequency range
    uint32_t upper_band_received_frequency;
    /// The number of frequency ids in the frame attached to this metadata
```
Since this code structure is never directly instantiated, there isn't much point in setting the values in the header, and it might just be confusing.  Better to just add a commend above that this structure should be initialized with the function below before use.

Reply:
Thanks; done.